homeassistant/components/climate/sensibo.py

```diff
@@ -0,0 +1,235 @@
+"""
+Support for Sensibo wifi-enabled home thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.sensibo/
+"""
+
+import asyncio
+import logging
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_API_KEY, CONF_ID, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+from homeassistant.components.climate import (
+    ATTR_CURRENT_HUMIDITY, ClimateDevice, PLATFORM_SCHEMA)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util.temperature import convert as convert_temperature
+
+REQUIREMENTS = ['pysensibo==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ALL = 'all'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_ID, default=ALL): vol.All(cv.ensure_list, cv.string),
+})
+
+_FETCH_FIELDS = ','.join([
+    'room{name}', 'measurements', 'remoteCapabilities',
+    'acState', 'connectionStatus{isAlive}'])
+_INITIAL_FETCH_FIELDS = 'id,' + _FETCH_FIELDS
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up Sensibo devices."""
+    import pysensibo
+
+    client = pysensibo.SensiboClient(
+        config[CONF_API_KEY], session=async_get_clientsession(hass))
+    devices = []
+    try:
+        for dev in (
+                yield from client.async_get_devices(_INITIAL_FETCH_FIELDS)):
```

Comment:
There is a default timeout on operations of 5 minutes: https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client.py#L37

I'll pass in lower timeout.

Would it be cleaner if the client session additionally had a default read and connect timeouts? https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client.py#L55